### DISCUSSION TITLE
Remove pending KYC user automatically

### DISCUSSION THREAD

**Discussion by leisterfrancisco** [eoscostarica/proton-affiliate#373] [Ideas]
# Improve UX

The next proposal is to get better handling for users that have applied to the Proton Affiliate program without `KYC`.

## Solution 
Create a backend worker service that will send emails to Proton Affiliate applicants that haven't completed the minimum requirement (`KYC`) to be accepted into the program and remove them if after a time they still haven't got the 'KYC'.
- After the first passed week, if the user hasn't completed the `KYC` requirement, the system will send a mail warning them about the missing requirement.
- After the first month, if the user hasn't completed the `KYC` requirement, the user will be deleted from the pending list with a final mail specifying the reason they will need to re-apply once they get `KYC`.

### Benefits from this solution
- The admin user shouldn't care about those users that didn't complete the KYC requirement after a while and the backend worker will send the mail to those users.
- Help admin users to focus only on the acceptance of those users that could belong to the Proton Affiliate program.

### Aditional solutions
- Create a table in the database to handle the time parameters and other kinds of future stuff that would help the admin user to control and modify easily without the need for a code update to get the modifications. For example:

| warning_users_by_kyc_after      | remove_users_by_kyc_after | other_parameters|
| ----------- | ----------- | ----------- |
| 1 week      | 1 month       | ...       |

**Comment by xavier506** [comment]
this solution looks good to me! However what happens when expiration times are adjusted in the smart contract ( we only recently changed to to be 30 days)

**Reply by leisterfrancisco** [reply]
We could adjust the logic time to work accordingly to the smart contract configuration
